### DISCUSSION TITLE
Android

### DISCUSSION THREAD

**Discussion by figmentdragon** [MicroPad/Web#319] [Ideas]
I just discovered MicroPad and love it.   The only thing, only program, that has kept me attached to Microsoft has been OneNote - and among the many many options, none of them felt intuitive, encapsulated a wide variety of options without the program feeling humongous and getting lost in navigation...  then i found the versatility of MicroPad.  Wow.  And it's truly cross platform?!  AMAZING.  

But....

There's no .apk for Android.  Is this on the horizon?  Please say yes.  Because if that's so, then i can completely cut free of most huge corporation invasive-ness.

**Comment by NickGeek** [comment]
Hi,
Thanks for the kind feedback! The whole reason this project originally was made was so a truly open alternative to OneNote could exist.

Mobile apps have been an ongoing problem for MicroPad that I've tried to address a few times. I prioritised tablet/desktop usage during the original design so some concepts are hard to translate.

# Past work
## Android Prototype (MicroPad Classic)
https://github.com/MicroPad/Android was the original attempt at porting MicroPad v1/v2 to mobile from 2016/2017, but it was never completed and isn't compatible with MicroPad v3:

### Notepad explorer (sidebar equivalent)
<img height="300" src="https://user-images.githubusercontent.com/3462055/113526296-a8765280-960d-11eb-9670-72dd554e1a1e.png" alt="sections and notes"/>

### Note viewer
<img height="300" src="https://user-images.githubusercontent.com/3462055/113526438-24709a80-960e-11eb-8294-04b8a5b8585b.png" alt="note initial position"/>
<img height="300" src="https://user-images.githubusercontent.com/3462055/113526437-233f6d80-960e-11eb-80af-5e3c5556503e.png" alt="note scrolled over"/>


## Cross-platform MicroPad Viewer
In 2019 I made a prototype cross-platform mobile app that was a limited, read-only MicroPad "viewer" called PicoPad:

### Notepad explorer
<img height="300" src="https://user-images.githubusercontent.com/3462055/113526517-77e2e880-960e-11eb-9e39-65ba73beeccd.png" alt="sections and notes"/>

### Note viewer
I managed to solve a lot of UX issues on mobile by representing notes one element at a time, instead of attempting to show the whole note on a phone screen.

<img height="300" src="https://user-images.githubusercontent.com/3462055/113526599-c2fcfb80-960e-11eb-9d52-f890283fad40.png" alt="note in initial position"/>
<img height="300" src="https://user-images.githubusercontent.com/3462055/113526594-c1cbce80-960e-11eb-90c7-efc3147c296f.png" alt="note scrolled over"/>

### Search
<img height="300" src="https://user-images.githubusercontent.com/3462055/113526234-5cc3a900-960d-11eb-8a13-81077107d800.png" alt="search results"/>
<img height="300" src="https://user-images.githubusercontent.com/3462055/113526231-5af9e580-960d-11eb-9bb8-c750466537c0.png" alt="No search results"/>

# Future work
I think the best approach for an Android app would be to grow PicoPad into a full MicroPad app, which is no small task.
Right now, I mostly use the MicroPad website on my phone (if you add it to your home screen it should work offline). When you dismiss the sidebar you get zoom controls. It's definitely somewhat janky though.

I can't offer any guarantees on how long it would take for PicoPad to be usable though.
